### PR TITLE
Add support for IDEF[], fixes #28 from upstream vttLib

### LINKED ISSUE #28: Support for IDEF instruction?
I'm looking at a font with VTT sources that use the `IDEF` instruction, and `vttLib.vtt_compile()` fails on that:

```
  File "xxx/venv/lib/python3.7/site-packages/vttLib/__init__.py", line 128, in split_functions
    assert 0, "Unexpected token in fpgm: %s" % t
AssertionError: Unexpected token in fpgm: ['IDEF', 145]
```

`IDEF` is very similar to `FDEF`, and I quickly tried a patch where everywhere it looks for `FDEF` it will also look for `IDEF`, but that must be too naive, and doesn't work.

Does anyone here have a hint (no pun intended) of how to implement this so I can try to make a PR after all? Although I'm afraid I might be over my head, because this is the first time I look at this code base and I don't know all that much about VTT to begin with.

### PR DESCRIPTION
**Pull request by robmck-ms**
Adds support for IDEF[]s similar to FDEF[]s. 

Closes #28

### CODE REVIEW

**Comment by belluzj** [comment]
Hello Rob, thanks for the PR. The changes look good to me. Could you please add a test? I think what you need to do is:

1. Add 2 files to the [test/data/](https://github.com/daltonmaag/vttLib/tree/master/tests/data) folder, for example called `idef_35.txt` and `idef_35_transformed.txt`
2. Add the filename of your new test to this list: https://github.com/daltonmaag/vttLib/blob/master/tests/vttLib_test.py#L11

You can probably start by adding a blank `_transformed` file, let the test fail and tell you what the output is, paste it in the `_transformed` file and check it then.

**Comment by robmck-ms** [comment]
Will do. 

I also have a second small change in the pipeline to support PUSHx[] and NPUSHx[] in a #PUSHOFF context. Do you want me to do that as a separate PR, or put it in with this one? I'll add tests for that too.

**Comment by robmck-ms** [comment]
Done adding the tests. Can add the PUSHOFF stuff here or do a separate PR, as you like.

**Comment by madig** [comment]
Thanks, will have a look next week. Please add separate changes to a separate PR.

File: src/vttLib/__init__.py [hunk 1]
```diff
@@ -137,14 +138,18 @@ def merge_functions(functions, include=None):
                     stack.append(num)
                 except ValueError:
                     break
-        if token.startswith("FDEF"):
+        if token.startswith("FDEF") or token.startswith("IDEF"):
```
Comment (madig) [inline src/vttLib/__init__.py]:
```suggestion
        if token.startswith(("FDEF", "IDEF")):
```

**Comment by madig** [comment]
Looks good otherwise, thanks! :)

**Comment by robmck-ms** [comment]
Good catch. Thanks!

**Pull request merged**